pkg/gadget-context/gadget-context.go

```diff
@@ -317,6 +317,9 @@ func (c *GadgetContext) LoadGadgetInfo(info *api.GadgetInfo, paramValues api.Par
 	}
 
 	if run {
+		if err := c.start(localOperators); err != nil {
+			return fmt.Errorf("starting local operators: %w", err)
+		}
```

Comment:
Maybe move the `start()` contents back into `run()` and create the go routine for the waiting/stopping inside of there as well (instead of below). Otherwise it's confusing that there's a `start()` and `run()` but no `stop()` or similar.